### PR TITLE
Manage Account Email Search. ACDM-1046 #resolve

### PR DESCRIPTION
**Pull request by pasadinhas**
Filter inactive & invalid email addresses, return all matches

### CODE REVIEW

File: src/main/java/org/fenixedu/academic/domain/Person.java [hunk 1]
```diff
@@ -1101,6 +1101,15 @@ public static Person readPersonByEmailAddress(final String email) {
         return emailAddress != null && emailAddress.getParty().isPerson() ? (Person) emailAddress.getParty() : null;
     }
 
+    public static Stream<Person> readPeopleByEmailAddress(final String email) {
```
Comment (jcarvalho) [inline src/main/java/org/fenixedu/academic/domain/Person.java]: I think the `Person` class is bloated enough as it is, I see no need to add a method that is basically a wrapper, and that is only invoked in a single place.

File: src/main/java/org/fenixedu/academic/domain/Person.java [hunk 2]
```diff
@@ -1101,6 +1101,15 @@ public static Person readPersonByEmailAddress(final String email) {
...
+        return EmailAddress.findAllActiveAndValid(email)
+                .filter(Objects::nonNull)
+                .map(emailAddress -> emailAddress.getParty())
```
Comment (jcarvalho) [inline src/main/java/org/fenixedu/academic/domain/Person.java]: You can just do `.map(EmailAddress::getParty)`.

File: src/main/java/org/fenixedu/academic/domain/Person.java [hunk 3]
```diff
@@ -1101,6 +1101,15 @@ public static Person readPersonByEmailAddress(final String email) {
...
+                .filter(Objects::nonNull)
+                .filter(party -> party.isPerson())
+                .map(party -> (Person) party);
```
Comment (jcarvalho) [inline src/main/java/org/fenixedu/academic/domain/Person.java]: Here you can do `.map(Person.class::cast)` instead.

**Comment by pasadinhas** [comment]
I fixed those issues.

**Pull request merged**